File: src/test/java/com/InkBamboo/Test/TestMain.java
package com.InkBamboo.Test;

import org.apache.flink.api.common.functions.FlatMapFunction;
import org.apache.flink.api.common.functions.MapFunction;
import org.apache.flink.streaming.api.datastream.DataStreamSource;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
import org.apache.flink.streaming.api.scala.DataStream;
import org.apache.flink.util.Collector;
import scala.Tuple2;
import scala.Tuple3;
import scala.Tuple4;

import java.util.ArrayList;

/**
 * Created By InkBamboo
 * Date: 2019/1/19 15:03
 * Calm Positive
 * Think Then Ask
 *
 * 测试通过，javaAPI获取数据源时使用的是DatastreamSource，在进行算子操作之后就会转变为Datastream了。
 */
public class TestMain {
    public static void main(String[] args) throws Exception {

        StreamExecutionEnvironment senv = StreamExecutionEnvironment.getExecutionEnvironment();

        Tuple3 tuple3 = new Tuple3("1","2","aas");
        ArrayList list = new ArrayList();
        list.add(tuple3);
        list.add(new Tuple3<String,String,String>("2","3","4"));
      DataStreamSource<Tuple3> streams =  senv.fromElements(new Tuple3("1","2","ass"),new Tuple3("2","3","bbc")).setParallelism(1);
     streams.map(new MapFunction<Tuple3, Tuple2>() {
         @Override
         public Tuple2 map(Tuple3 value) throws Exception {
             return new Tuple2(value._1(),value);
         }
     })
     .flatMap(new FlatMapFunction<Tuple2, wordcount>() {
         @Override
         public void flatMap(Tuple2 value, Collector<wordcount> out) throws Exception {
             Tuple3 tpl3 = (Tuple3) value._2();
             out.collect(new wordcount(value._1.toString(),Integer.parseInt(tpl3._1().toString()),tpl3._2().toString(),tpl3._3().toString()));
         }
     }).keyBy("colOne")
             .sum("colTwo")
     .print();

     senv.execute("test Main");
    }
}

File: src/test/scala/TestMain.scala

import net.sf.json.JSONObject
import org.joda.time.DateTime
import org.apache.flink.api.scala._
import org.apache.flink.streaming.api.TimeCharacteristic
import org.apache.flink.streaming.api.scala.StreamExecutionEnvironment
import org.apache.flink.streaming.api.windowing.time.Time
/**
  * Created By InkBamboo
  * Date: 2018/12/30 16:59
  * Calm Positive
  * Think Then Ask
  */
object TestMain extends App {

  //1. json字符串封装成类对象
  val str = "{\"word\":\"a3bcd\",\"time\":1546146333716,\"count\":1}"
  val jSONObject = JSONObject.fromObject(str)

  val word =JSONObject.toBean(jSONObject,wordcount2.getClass)

  println("-------------------2-------------------------")
  //2. string类型转换为DateTime

  val str2= "2019-01-23T14:37:07.282+08:00"
  val dateTime = new DateTime(str2)

  println(dateTime)

}
case class wordcount2(word:String,time:Long,count:Int)


//maxby取最大值
object StreamTest extends App {

  val senv = StreamExecutionEnvironment.getExecutionEnvironment
  senv.setStreamTimeCharacteristic(TimeCharacteristic.EventTime)
 // senv.setParallelism(1)
 val element =  senv.fromElements(new Tuple3("a",2,1511658000),new Tuple3("b",4,1511658000),new Tuple3("a",5,1511658000),new Tuple3("b",2,1511658000))
  element
      .assignAscendingTimestamps(x=>x._3)
    .keyBy(0)
    .timeWindow(Time.seconds(10))
    .maxBy(1)
    //.sum(1)
    .print()

  senv.execute("testMaxBy")
}